openpype/client/entity_links.py

```diff
@@ -127,12 +128,20 @@ def get_linked_representation_id(
     if not version_id:
         return []
 
+    version_doc = get_version_by_id(
+        project_name, version_id, fields=["type", "version_id"]
+    )
+    if version_doc["type"] == "hero_version":
+        version_id = version_doc["version_id"]
+
     if max_depth is None:
         max_depth = 0
 
     match = {
         "_id": version_id,
-        "type": {"$in": ["version", "hero_version"]}
+        # Links are not stored to hero versions at this moment so filter
+        #   is limited to just versions
+        "type": "version"
```

Comment:
Actually - are we sure? The input link collection I believe is retrieved from containers in the scene - those I suspect would have the representation id of the hero version. As such by doing this "fix" we'd never be able to list those inputs even though they are stored?

Reply:
Links are added only to [versioned version documents](https://github.com/pypeclub/OpenPype/blob/develop/openpype/plugins/publish/integrate_inputlinks.py#L45). and [hero version document](https://github.com/pypeclub/OpenPype/blob/develop/openpype/plugins/publish/integrate_hero_version.py#L195) don't have `"data"` at all.

But because hero versions are based on versioned documents we can reuse links from there.

Reply:
Yes. Getting inputs FROM the hero version is then fixed. But any hero version IN the inputs is not? That hero version's inputs are not collected? (But might not be relevant for the UI we have no since it only shows direct inputs instead of a further depth.)

That does raise the question too that when we collect the inputs on publish would we want to resolve the hero version to the exact version it actually was at the time or do we want to just keep the link to the hero version? If we'd 'bake' it down to the exact version on collecting inputs the issue I mentioned would be gone plus you get the benefit of actually being able to see what version was used at the time in input link. Maybe we could tag it extra as "hero" so that the does know it originally was a hero link but at the time of publishing that was referring to v038.

@mkolar thoughts?

Reply:
Should we spin this up as a new issue?